File: code/community/1306408209740730371/whisper-3-nova-3-deepgram-enhancements.py
# Import necessary libraries
import os
import requests

# Define the audio file path and API details
AUDIO_FILE_PATH = 'your_audio_file_path.wav'
DEEPGRAM_API_URL = "https://api.deepgram.com/v1/listen"
DEEPGRAM_API_KEY = os.environ.get('DEEPGRAM_API_KEY')

# Ensure the API key is set
if DEEPGRAM_API_KEY is None:
    raise ValueError("The DEEPGRAM_API_KEY environment variable must be set.")

# Read audio file
with open(AUDIO_FILE_PATH, 'rb') as audio_file:
    audio_data = audio_file.read()

# Set headers for the request
headers = {
    'Authorization': f'Token {DEEPGRAM_API_KEY}',
    'Content-Type': 'audio/wav'
}

# Make post request to Deepgram API
response = requests.post(DEEPGRAM_API_URL, headers=headers, data=audio_data)

# Check if request was successful
if response.status_code == 200:
    # Do something with the response, for example log the transcription
    result = response.json()
    print("Transcription results:", result)
else:
    # Handle the error
    print(f"Error: {response.status_code} - {response.text}")

File: code/community/1306408209740730371/whisper-3-nova-3-deepgram-enhancements.js
// Import the required modules
const fs = require('fs');
const fetch = require('node-fetch');

// Define environment variables
const API_KEY = process.env.DEEPGRAM_API_KEY;
const FILE_PATH = 'your_audio_file_path.wav';

// Ensure API Key is available
if (!API_KEY) {
    console.error('DEEPGRAM_API_KEY environment variable must be set.');
    process.exit(1);
}

// Read audio file
const audioData = fs.readFileSync(FILE_PATH);

// Prepare request options
const options = {
    method: 'POST',
    headers: {
        'Authorization': `Token ${API_KEY}`,
        'Content-Type': 'audio/wav',
    },
    body: audioData
};

// Function to transcribe audio
async function transcribeAudio() {
    try {
        const response = await fetch('https://api.deepgram.com/v1/listen', options);
        const data = await response.json();

        if (response.ok) {
            console.log('Transcription results:', data);
        } else {
            console.error(`Error: ${response.status} - ${data.message}`);
        }
    } catch (error) {
        console.error('Error:', error);
    }
}

transcribeAudio();